crates/sui-core/src/checkpoints/checkpoint_executor/data_ingestion_handler.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::authority::AuthorityStore;
+use crate::checkpoints::CheckpointStore;
+use std::collections::{HashMap, HashSet};
+use std::path::PathBuf;
+use std::sync::Arc;
+use sui_types::digests::TransactionDigest;
+use sui_types::effects::TransactionEffectsAPI;
+use sui_types::error::{SuiError, SuiResult, UserInputError};
+use sui_types::full_checkpoint_content::{CheckpointData, CheckpointTransaction};
+use sui_types::messages_checkpoint::VerifiedCheckpoint;
+use sui_types::storage::ObjectKey;
+
+pub(crate) fn store_checkpoint_locally(
+    path: PathBuf,
+    checkpoint: VerifiedCheckpoint,
+    authority_store: Arc<AuthorityStore>,
+    checkpoint_store: Arc<CheckpointStore>,
+    transaction_digests: Vec<TransactionDigest>,
+) -> SuiResult {
```

Comment:
This works to get started, but i wonder if there is a way to leverage the fact that checkpoint executor should have already loaded a majority of this data from the database, saving us the need to load it again...although maybe rocksdb caches handle this for us nicely already.

Reply:
This is fresh data, so rocksdb should have all this data in the memtable. 
But yeah, it would be nice to just pass it down instead. 
We've already saved a couple of reads (compared to the REST API implementation) by passing checkpoint and transaction digests. 
I don't see any other obvious savings with the current checkpoint executor semantics, but we can follow up on that